staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -97,3 +97,25 @@ type HTTPConnectConfig struct {
 	// +optional
 	ClientCert string
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// LeaderMigratorConfiguration provides versioned configuration for all migrating leader locks.
+type LeaderMigratorConfiguration struct {
```

Comment:
Whats the reason for putting this in k8s.io/apiserver?  Instead of, say, client-go.

Reply:
Going to defer to @cheftako on this one since he recommended we use this API group. 

The thinking is that this a component level type and not something we want to be exposing at the client level.

Reply:
When this was written we really didn't have a good home for this. At this point it probably belongs in staging/controller-manager